File: 2253.c
#include <stdio.h>
#include <string.h>

int main(void)
{
    char senha[50];
    while (scanf("%[^\n]s", senha) != EOF)
    {
        getchar();
        short tam = strlen(senha), validar = 1, maius = 0, minus = 0;
        
        if (tam < 6 || tam > 32)
            validar = 0;
        else
        {
            int i;
            for (i = 0; i < tam; i++)
            {
                if (senha[i] >= 'a' && senha[i] <= 'z')
                    maius = 1;
                else
                {
                    if (senha[i] >= 'A' && senha[i] <= 'Z')
                        minus = 1;
                    else
                        if (senha[i] >= '0' && senha[i] <= '9')
                            validar = 1;
                        else
                        {
                            validar = 0;
                            break;
                        }
                }
            }  
        }
        printf(validar == 1 && minus == 1 && maius == 1? "Senha valida.\n" : "Senha invalida.\n");
    }
    return 0;
}

File: 2253.py

from string import ascii_lowercase as lowercase, ascii_uppercase as uppercase, digits

while True:
    try:
        senha = input()
    except EOFError:
        break

    validar, minus, maius = True, False, False
    
    tam = len(senha)
    if tam < 6 or tam > 32:
        validar = False
    else:
        for i in senha:
            if i not in digits + uppercase + lowercase:
                validar = False
                break
            elif not minus or not maius:
                if i in lowercase and minus == False:
                    minus = True
                elif i in uppercase and maius == False:
                    maius = True

    if validar == True and minus == True and maius == True:
        print('Senha valida.')
    else:
        print('Senha invalida.')
